packages/core/src/interpreter.ts

```diff
@@ -303,12 +311,10 @@ export class Interpreter<
     registry.register(this.sessionId, this.ref);
     this.status = ActorStatus.Running;
 
-    let resolvedState = initialState
-      ? this.behavior.restoreState?.(initialState, this._actorContext)
-      : this._getInitialState() ?? undefined;
+    let resolvedState = this._getInitialState();
 
     if (this.behavior.start) {
-      resolvedState = this.behavior.start(resolvedState, this._actorContext);
+      this.behavior.start(resolvedState, this._actorContext);
```

Comment:
I wonder if there are any unwanted things that might happen if we call `.start()` with a rehydrated state. How do we describe the exact role of `.start()`?

Reply:
The role of `.start(state)` is to _attempt_ to put an actor at the specified `state`.

Reply:
In other words - it's used both for starting a fresh actor and for starting a rehydrated one. I think that overall this is OK - but the implementers need to be aware of this.